assets/js/googlesitekit/api/middleware/preloading.test.js

```diff
@@ -0,0 +1,179 @@
+		const requestURI = 'google-site-kit/v1/core/user/authentication';
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData
+		);
+
+		const requestOptions = {
+			method: 'GET',
+			path: requestURI,
+		};
+		const next = jest.fn();
+		const firstResponse = await preloadingMiddleware( requestOptions, next );
+		expect( firstResponse ).toEqual( preloadedData[ requestURI ].body );
+		expect( next ).not.toHaveBeenCalled();
+	} );
+
+	it( 'returns a preloaded response from multiple URIs', async () => {
+		const firstRequestURI = 'google-site-kit/v1/core/user/authentication';
+		const secondRequestURI = 'google-site-kit/v1/core/site/data/connection';
+
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData
+		);
+
+		const firstResponseNext = jest.fn();
+		const secondResponseNext = jest.fn();
+
+		const firstResponse = await preloadingMiddleware( { method: 'GET', path: firstRequestURI }, firstResponseNext );
+		expect( firstResponse ).toEqual( preloadedData[ firstRequestURI ].body );
+		expect( firstResponseNext ).not.toHaveBeenCalled();
+
+		const firstResponseAgain = await preloadingMiddleware( { method: 'GET', path: firstRequestURI }, firstResponseNext );
+		expect( firstResponseAgain ).toBeUndefined();
+		expect( firstResponseNext ).toHaveBeenCalled();
+
+		const secondResponse = await preloadingMiddleware( { method: 'GET', path: secondRequestURI }, secondResponseNext );
+		expect( secondResponse ).toEqual( preloadedData[ secondRequestURI ].body );
+		expect( secondResponseNext ).not.toHaveBeenCalled();
+
+		const secondResponseAgain = await preloadingMiddleware( { method: 'GET', path: secondRequestURI }, secondResponseNext );
+		expect( secondResponseAgain ).toBeUndefined();
+		expect( secondResponseNext ).toHaveBeenCalled();
+	} );
+
+	it( 'does nothing and calls next middleware when no preloaded response exists for the request', async () => {
+		const requestURI = 'google-site-kit/v1/core/user/authentication';
+		const next = jest.fn();
+		const preloadingMiddleware = createPreloadingMiddleware( {} );
+		const requestOptions = {
+			method: 'GET',
+			path: requestURI,
+		};
+		await preloadingMiddleware( requestOptions, next );
+		expect( next ).toHaveBeenCalled();
+	} );
+
+	it( 'returns a preloaded response only once for each preloaded request', async () => {
+		const requestURI = 'google-site-kit/v1/core/user/authentication';
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData
+		);
+
+		const requestOptions = {
+			method: 'GET',
+			path: requestURI,
+		};
+		const next = jest.fn();
+		const firstResponse = await preloadingMiddleware( requestOptions, next );
+		expect( firstResponse ).toEqual( preloadedData[ requestURI ].body );
+		expect( next ).not.toHaveBeenCalled();
+
+		const secondRequest = await preloadingMiddleware( requestOptions, next );
+		expect( secondRequest ).toBeUndefined();
+		expect( next ).toHaveBeenCalled();
+	} );
```

Comment:
Added a blank line between tests 👍 
```suggestion
	} );

```